phoenix-core/src/main/java/org/apache/phoenix/compile/ScanRanges.java

```diff
@@ -615,9 +616,9 @@ private static TimeRange getRowTimestampColumnRange(List<List<KeyRange>> ranges,
                 if (ranges != null && ranges.size() > rowTimestampColPos) {
                     List<KeyRange> rowTimestampColRange = ranges.get(rowTimestampColPos);
                     List<KeyRange> sortedRange = new ArrayList<>(rowTimestampColRange);
-                    Collections.sort(sortedRange, KeyRange.COMPARATOR);
-                    //ranges.set(rowTimestampColPos, sortedRange); //TODO: do I really need to do this?
                     Field f = schema.getField(rowTimestampColPos);
+                    Collections.sort(sortedRange, f.getSortOrder() == SortOrder.ASC ? KeyRange.COMPARATOR : KeyRange.DESC_COMPARATOR);
+                    //ranges.set(rowTimestampColPos, sortedRange); //TODO: do I really need to do this?
```

Comment:
Do you think it makes sense to remove this commented out code now?

Reply:
Its not needed will remove it.